### PR TITLE
RepositoryResolver.resolveAsSet fix

### PR DESCRIPTION
**Pull request by Azquelt**
* When resolving as a set, report visibility correctly as this affects how rules for dependency toleration are applied
* Update existing unit tests
* Update resolution tests adapted from the failure cases provided by @TayyabDev 
* Add conflict chain information to `RepositoryResolutionException`
* Update the message for `RepositoryResolutionException` to aid debugging

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Build failed due to an infrastructure error. The repository tests did run and pass.

File: dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java [hunk 1]
```diff
@@ -1912,11 +1914,179 @@ public void testCircleNotToRoot() throws Exception {
...
+    }
+
+    /**
+     * Test the resolveAsSet method with 3 level feature dependencies
+     *
+     * All features are public
+     *
+     * Dependency map
+     * Feature A -> Feature X -> Feature I 1.0
+     * Feature B -> Feature Y -> Feature I 1.1 tolerate 1.0
```
Comment (pnickoll) [inline dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java]: Is this comment wrong? Implies Feature B tolerates 1.0 when it doesn't?

File: dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java [hunk 2]
```diff
@@ -1912,11 +1914,179 @@ public void testCircleNotToRoot() throws Exception {
...
+    }
+
+    /**
+     * Test the resolveAsSet method with 3 level feature dependencies
+     *
+     * Feautre I is private
+     *
+     * Dependency map
+     * Feature A -> Feature X -> Feature I 1.0
+     * Feature B -> Feature Y -> Feature I 1.1 tolerate 1.0
```
Comment (pnickoll) [inline dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java]: Same as above, wrong comment?

File: dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java [hunk 3]
```diff
@@ -1912,11 +1914,179 @@ public void testCircleNotToRoot() throws Exception {
...
+
+        // build the repository and resolve the features
+        RepositoryResolver resolver = createResolver();
+        Collection<List<RepositoryResource>> resolved = resolver.resolveAsSet(Arrays.asList(featureA.getProvideFeature(), featureB.getProvideFeature()));
+        assertThat(resolved, containsInAnyOrder(
+                                                Matchers.<RepositoryResource> contains(featureI10, featureX, featureA),
+                                                Matchers.<RepositoryResource> contains(featureI10, featureY, featureB)));
+
+    }
+
```
Comment (pnickoll) [inline dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java]: Is it worth also having a test where 
     * Dependency map
     * Feature A -> Feature X -> Feature I 1.1
     * Feature B -> Feature Y -> Feature I 1.0 tolerate 1.1

I.e. B uses an older level but tolerates a newer level, does this make any difference to the logic being tested?

File: dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java (hunk 1 above)
Comment (pnickoll) [reply dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java]: Spoke to @anrouse - I misread the comment, consider this resolved...

File: dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java (hunk 2 above)
Comment (pnickoll) [reply dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java]: Spoke to @anrouse and I have misread, consider this resolved

File: dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java (hunk 3 above)
Comment (Azquelt) [reply dev/com.ibm.ws.repository.resolver/test/com/ibm/ws/repository/resolver/ResolutionTests.java]: Yeah, I can add this

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
#build

**Pull request merged**